doc/source/whatsnew/v0.19.0.txt

```diff
@@ -878,6 +878,31 @@ of ``int64`` (:issue:`13988`)
    pi = pd.PeriodIndex(['2011-01', '2011-02'], freq='M')
    pi.values
 
+.. _whatsnew_0190.api.setops:
+
+Index ``+`` / ``-`` no longer used for set operations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Addition and subtraction of certain Index types performed set operations
+(set union and difference). This behaviour was already deprecated since 0.15.0,
+and is now disabled and, when possible, ``+``
+and ``-`` are used for numeric operations (:issue:`8227`, :issue:`14127`).
```

Comment:
I would be more specific that 'when possible' (maybe say for numeric & datetimelike)